hbase-server/src/main/java/org/apache/hadoop/hbase/replication/HBaseReplicationEndpoint.java

```diff
@@ -148,27 +149,30 @@ public boolean isAborted() {
     return false;
   }
 
+  /**
+   * Get the connection to peer cluster
+   * @return connection to peer cluster
+   * @throws IOException
+   */
+  protected synchronized Connection getPeerConnection() throws IOException {
+    if (peerConnection == null) {
+      peerConnection = ConnectionFactory.createConnection(ctx.getConfiguration());
+    }
+    return peerConnection;
+  }
+
   /**
    * Get the list of all the region servers from the specified peer
-   * @param zkw zk connection to use
    * @return list of region server addresses or an empty list if the slave is unavailable
    */
-  protected static List<ServerName> fetchSlavesAddresses(ZKWatcher zkw)
```

Comment:
Should be compatible with the old ZK impl. Cannot remove them directly?

Reply:
OK，try new impl firstly and old ZK impl secondly, is that OK?